### DISCUSSION TITLE
Database

### DISCUSSION THREAD

**Discussion by vladtopchu** [serbelga/Todometer-KMP#446] [Q&A] [answered]
Hi! I'm new to KMP and SQDelight in particular, so I can't really understand how TodometerDatabase works
I've tried to remove "database" folder in common.database (where TodometerDatabase stores) and it is generating again and I wonder what is responsible for it? 
Is it sqldelight or is it some custom behaviour?

**Comment by pablichjenkov** [comment]
The SQLDelight plugin generates it automatically.

**Reply by vladtopchu** [reply]
but how does plugin knows what name to give to database class (TodometerDatabase in our case)?

**Comment by vladtopchu** [comment, marked as answer]
ok, i found it! it's specified in build.gradle inside common.database module :)
